ProjectPaula/src/ProjectPaula/DAL/PaulRepository.cs

```diff
@@ -253,39 +253,68 @@ public static Task<List<CourseCatalog>> GetCourseCataloguesAsync(DatabaseContext
         /// <returns>Task</returns>
         public static async Task UpdateAllCoursesAsync()
         {
-            _isUpdating = true;
-            UpdateStarting?.Invoke();
-            var parser = new PaulParser();
-            using (var context = new DatabaseContext(_filename, _basePath))
+            if (!_isUpdating)

```

Comment:
This isn't thread-safe and should be done using `Interlocked`: https://stackoverflow.com/a/29180781/1502352